ts/lib/src/index.ts

```diff
@@ -2,84 +2,163 @@ import * as secp from "@noble/secp256k1";
 
 export type Keypair = { sk: Uint8Array; pk: Uint8Array };
 
+/** * Generates a new keypair using secp256k1.
+ * The private key (sk) is a random 32-byte Uint8Array,
+ * and the public key (pk) is derived from the private key.
+ * @returns A Keypair object containing the private key (sk) and public key (pk).
+ */
 export function generateKeypair(): Keypair {
   const sk = secp.utils.randomPrivateKey();
   const pk = secp.getPublicKey(sk, true);
   return { sk, pk };
 }
 
+/** * Converts a private key (sk) to a public key (pk).
+ * The private key can be provided as a Uint8Array or a hex string.
+ * @param sk The private key to convert. It can be a Uint8Array or a hex string.
+ * @returns The public key as a Uint8Array.
+ * @throws Error if the private key length is not 32 bytes.
+ */
+export function publicKeyFromPrivateKey(sk: Uint8Array | string): Uint8Array {
+  const privateKey = ensureBytes(sk);
+  if (privateKey.length !== 32) {
+    throw new Error("Invalid private key length, expected 32 bytes");
+  }
+  return secp.getPublicKey(privateKey, true);
+}
+
+/**
+ * Gets the appropriate Crypto API to use.
+ * @returns The global Crypto API or the Node.js crypto module.
+ * If the global Crypto API is not available, it will import the Node.js crypto module.
+ */
 export async function getCrypto(): Promise<Crypto> {
   return typeof globalThis.crypto !== "undefined"
-  ? globalThis.crypto
-  : ((await import("node:crypto")).webcrypto as Crypto);
+    ? globalThis.crypto
+    : ((await import("node:crypto")).webcrypto as Crypto);
 }
 
-export function toHex(uint8: Uint8Array, withPrefix: boolean = false): string {
-  const hex = Array.from(uint8)
-  .map((byte) => byte.toString(16).padStart(2, "0"))
-  .join("");
+/**
+ * Converts a Uint8Array to a hex string.
+ * @param bytes The Uint8Array to convert.
+ * @param withPrefix If true, the hex string will be prefixed with "0x".
+ * @returns The hex string representation of the Uint8Array.
+ */
+export function toHexString(
```

Comment:
Assuming [semver](https://semver.org/)  convention, this is actually a breaking change, so we should bump to 0.2.0 here. This does not have that much difference since we don't publish this to npm, yet I'd stick to the convention anyway.